lib/savepoint.js

```diff
@@ -3,14 +3,13 @@ function randomName() {
 }
 
 module.exports = function(dialect) {
-    return function() {
+    return function beginSavepoint() {
         var tx = this;
-        var spname = randomName();
-        if (dialect == 'mysql')
-            spname = '`' + spname + '`';
-        else
-            spname = '"' + spname + '"';
-        tx.query('SAVEPOINT ' + spname);
+        let rname = randomName()
+        var spname = dialect == 'mysql' ? '`' + rname + '`': rname = '"' + rname + '"';
+
+        const ready = tx.queryAsync('SAVEPOINT ' + spname)
+
         function restore(cb) {
             return tx.query('ROLLBACK TO SAVEPOINT ' + spname, cb);
```

Comment:
Can you take a look at this? https://www.postgresql.org/docs/9.3/static/plpgsql-control-structures.html#PLPGSQL-ERROR-TRAPPING

It may be useful to prevent the outer transaction from failing when the query is intentionally bad, eg: bad UUID, constraint validation. These may also be test scenarios, and we don't want the outer transaction to fail.